In [1]:
import pandas as pd

df = pd.read_csv("E:\\Download\\data.csv")
#In interactive environments, displays DataFrame in a potentially more formatted way.
df

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]

In [2]:
import pandas as pd
df = pd.read_csv("E:\\Download\\data.csv")
print(df.to_string()) 

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

In [5]:
df = pd.read_csv("E:\\Download\\data.csv", nrows=5)
df

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0

In [6]:
df = pd.read_csv("E:\\Download\\data.csv", nrows=1)
df

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1

In [3]:
df = pd.read_csv("E:\\Download\\data.csv")
print(type(df))

<class 'pandas.core.frame.DataFrame'>


In [24]:
df = pd.read_csv("E:\\Download\\data.csv", usecols=["Calories"])
print(df.to_string())

     Calories
0       409.1
1       479.0
2       340.0
3       282.4
4       406.0
5       300.0
6       374.0
7       253.3
8       195.1
9       269.0
10      329.3
11      250.7
12      345.3
13      379.3
14      275.0
15      215.2
16      300.0
17        NaN
18      323.0
19      243.0
20      364.2
21      282.0
22      300.0
23      246.0
24      334.5
25      250.0
26      241.0
27        NaN
28      280.0
29      380.3
30      243.0
31      180.1
32      299.0
33      223.0
34      361.0
35      415.0
36      300.0
37      300.0
38      300.0
39      266.0
40      180.1
41      286.0
42      329.4
43      400.0
44      397.0
45      273.0
46      387.6
47      300.0
48      298.0
49      397.6
50      380.2
51      643.1
52      263.0
53      486.0
54      238.0
55      450.7
56      413.0
57      305.0
58      226.4
59      321.0
60     1376.0
61     1034.4
62      853.0
63      341.0
64      131.4
65      800.4
66      873.4
67      816.0
68      110.4
69     1500.2
70    

The loop (for y in x) directly prints each value, so only the values appear, not the indices.
Single Column: When you are dealing with only one column (like in your last example with just Calories), dropna() automatically applies to that column, so no subset is necessary.

In [26]:
df = pd.read_csv("E:\\Download\\data.csv", usecols=["Calories"])
x = df["Calories"].dropna()
for y in x:
    print(y)

409.1
479.0
340.0
282.4
406.0
300.0
374.0
253.3
195.1
269.0
329.3
250.7
345.3
379.3
275.0
215.2
300.0
323.0
243.0
364.2
282.0
300.0
246.0
334.5
250.0
241.0
280.0
380.3
243.0
180.1
299.0
223.0
361.0
415.0
300.0
300.0
300.0
266.0
180.1
286.0
329.4
400.0
397.0
273.0
387.6
300.0
298.0
397.6
380.2
643.1
263.0
486.0
238.0
450.7
413.0
305.0
226.4
321.0
1376.0
1034.4
853.0
341.0
131.4
800.4
873.4
816.0
110.4
1500.2
1115.0
387.6
700.0
953.2
304.0
563.2
251.0
300.0
500.4
1729.0
319.2
344.0
151.1
500.0
225.3
300.0
234.0
1000.1
242.0
50.3
600.1
105.3
50.5
127.4
229.4
128.2
244.2
188.2
604.1
77.7
500.0
500.0
500.4
92.7
124.0
800.3
86.2
500.3
1860.4
325.2
275.0
124.2
225.3
367.6
351.7
443.0
277.4
332.7
193.9
100.7
336.7
344.9
368.5
271.0
275.3
382.0
466.4
384.0
342.5
357.5
335.0
327.5
339.0
189.0
317.7
318.0
308.0
222.4
390.0
250.4
335.4
470.2
270.8
400.0
361.9
185.0
409.4
343.0
353.2
374.0
275.8
328.0
368.5
270.4
270.4
382.8
240.9
250.4
260.4
270.0
280.9
290.8
300.0
310.2
320.4
330.4


Two or More Columns: When you have multiple columns but want to drop rows based on all those columns, you can call dropna() without subset since it checks all specified columns by default.

In [27]:
df = pd.read_csv("E:\\Download\\data.csv", usecols=["Calories", "Pulse"])
x = df.dropna()
print(x.to_string())

     Pulse  Calories
0      110     409.1
1      117     479.0
2      103     340.0
3      109     282.4
4      117     406.0
5      102     300.0
6      110     374.0
7      104     253.3
8      109     195.1
9       98     269.0
10     103     329.3
11     100     250.7
12     106     345.3
13     104     379.3
14      98     275.0
15      98     215.2
16     100     300.0
18     103     323.0
19      97     243.0
20     108     364.2
21     100     282.0
22     130     300.0
23     105     246.0
24     102     334.5
25     100     250.0
26      92     241.0
28     100     280.0
29     102     380.3
30      92     243.0
31      90     180.1
32     101     299.0
33      93     223.0
34     107     361.0
35     114     415.0
36     102     300.0
37     100     300.0
38     100     300.0
39     104     266.0
40      90     180.1
41      98     286.0
42     100     329.4
43     111     400.0
44     111     397.0
45      99     273.0
46     109     387.6
47     111     300.0
48     108   

In [29]:
df1 = pd.read_csv("E:\\Download\\data.csv", nrows=5, usecols=[1,2,3])
print(df1)

   Pulse  Maxpulse  Calories
0    110       130     409.1
1    117       145     479.0
2    103       135     340.0
3    109       175     282.4
4    117       148     406.0


In [7]:
import pandas as pd
df1 = pd.read_csv("E:\\Download\\data.csv", skiprows=[0,1])
print(df1)

     60  117  145  479.0
0    60  103  135  340.0
1    45  109  175  282.4
2    45  117  148  406.0
3    60  102  127  300.0
4    60  110  136  374.0
..   ..  ...  ...    ...
162  60  105  140  290.8
163  60  110  145  300.0
164  60  115  145  310.2
165  75  120  150  320.4
166  75  125  150  330.4

[167 rows x 4 columns]


In [32]:
import pandas as pd
df1 = pd.read_csv("E:\\Download\\data.csv", index_col="Pulse")
df1

Duration  Maxpulse  Calories
Pulse                              
110          60       130     409.1
117          60       145     479.0
103          60       135     340.0
109          45       175     282.4
117          45       148     406.0
...         ...       ...       ...
105          60       140     290.8
110          60       145     300.0
115          60       145     310.2
120          75       150     320.4
125          75       150     330.4

[169 rows x 3 columns]

In [13]:
df1 = pd.read_csv("E:\\Download\\data.csv", header = 4)
df1

45  109  175  282.4
0    45  117  148  406.0
1    60  102  127  300.0
2    60  110  136  374.0
3    45  104  134  253.3
4    30  109  133  195.1
..   ..  ...  ...    ...
160  60  105  140  290.8
161  60  110  145  300.0
162  60  115  145  310.2
163  75  120  150  320.4
164  75  125  150  330.4

[165 rows x 4 columns]

In [44]:
import pandas as pd
df2 = pd.read_csv("E:\\Download\\data.csv", names =["col1", "col2", "col3", "col4"])
df2

col1   col2      col3      col4
0    Duration  Pulse  Maxpulse  Calories
1          60    110       130     409.1
2          60    117       145       479
3          60    103       135       340
4          45    109       175     282.4
..        ...    ...       ...       ...
165        60    105       140     290.8
166        60    110       145       300
167        60    115       145     310.2
168        75    120       150     320.4
169        75    125       150     330.4

[170 rows x 4 columns]

Decimal Values: The values in the "Calories" column (like 409.1, 479, and 282.4) cannot be converted to integers directly because integers cannot represent decimal numbers.

In [56]:
df2 = pd.read_csv("E:\\Download\\data.csv", dtype={"Duration":"float"}) #convert int to float
df2

Duration  Pulse  Maxpulse  Calories
0        60.0    110       130     409.1
1        60.0    117       145     479.0
2        60.0    103       135     340.0
3        45.0    109       175     282.4
4        45.0    117       148     406.0
..        ...    ...       ...       ...
164      60.0    105       140     290.8
165      60.0    110       145     300.0
166      60.0    115       145     310.2
167      75.0    120       150     320.4
168      75.0    125       150     330.4

[169 rows x 4 columns]

If you have a DataFrame with multiple columns but want to focus on dropping rows only based on NaNs in specific columns, then you would use subset. This allows you to ignore NaNs in other columns.

Using df2 makes the code more efficient and straightforward, while a new variable could unnecessarily complicate things.

In [71]:
df2 = pd.read_csv("E:\\Download\\data.csv")
df2 = df2.dropna(subset=["Calories"])
df2["Calories"] = df2["Calories"].astype(int)
print(df2)

     Duration  Pulse  Maxpulse  Calories
0          60    110       130       409
1          60    117       145       479
2          60    103       135       340
3          45    109       175       282
4          45    117       148       406
..        ...    ...       ...       ...
164        60    105       140       290
165        60    110       145       300
166        60    115       145       310
167        75    120       150       320
168        75    125       150       330

[164 rows x 4 columns]
